templates/service_templates/redis.service.erb

```diff
@@ -1,15 +1,19 @@
 [Unit]
-Description=Advanced key-value store for <%= @title %>
+Description=Redis Advanced key-value store for instance <%= @title %>
 After=network.target
+After=network-online.target
+Wants=network-online.target
 
 [Service]
 Type=forking
-ExecStart=/usr/bin/redis-server <%= @redis_file_name %>
+ExecStart=/usr/bin/redis-server <%= @redis_file_name %> --supervised systemd
 ExecStop=/usr/bin/redis-server -p <%= @port %> shutdown
-TimeoutStopSec=0
 Restart=always
 User=<%= @service_user %>
 Group=<%= @service_user %>
+RuntimeDirectory=redis
+RuntimeDirectoryMode=0755
```

Comment:
A lot is changing here. Could you add a spec test that shows the contents of this file is being rendered appropriately.

Here's an example - https://github.com/ghoneycutt/puppet-module-bind/blob/master/spec/defines/key_spec.rb#L26-L50

Reply:
Added a spec test for the systemd service file.